spec/rubocop/cop/performance/use_zip_to_wrap_array_contents_spec.rb

```diff
@@ -0,0 +1,288 @@
+      expect_offense(<<~RUBY)
+        [nil, tuple].flatten.map { |e| [e] }.call
+                             ^^^^^^^^^^^^^^^ Use `.zip` instead of `.map { |id| [id] }`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        [nil, tuple].flatten.zip.call
+      RUBY
+    end
+  end
+
+  context 'when the map block does not contain an array literal' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].map { |id| id }
+      RUBY
+    end
+  end
+
+  context 'when using collect' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].collect { |id| [id] }
+      RUBY
+    end
+  end
+
+  context 'when using select with an array literal' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].select { |id| [id] }
+      RUBY
+    end
+  end
+
+  context 'when making an array of arrays using each_with_object' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1,2,3].each_with_object([]) {|id, object| object << [id]}
+      RUBY
+    end
+  end
+
+  context 'when using map with an array literal containing multiple elements' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].map { |id| [id, id] }
+      RUBY
+    end
+  end
+
+  context 'when using map with doubly wrapped arrays' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].map { |id| [[id]] }
+      RUBY
+    end
+  end
+
+  context 'when using map with addition' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].map { |id| id + 1 }
+      RUBY
+    end
+  end
+
+  context 'when using map with array addition' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].map { |id| [id] + [id] }
+      RUBY
+    end
+  end
+
+  context 'when using map with indexing into an array' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].map { |id| [id][id] }
+      RUBY
+    end
+  end
+
+  context 'when calling map with no arguments' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].map
+      RUBY
+    end
+  end
+
+  context 'when calling map with an empty block' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].map {}
+      RUBY
+    end
+  end
+
+  context 'when calling filter_map' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].filter_map {|id| [id]}
+      RUBY
+    end
+  end
+
+  context 'when calling flat_map' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].flat_map {|id| [id]}
+      RUBY
+    end
+  end
+
+  context 'when using Array.wrap' do
+    it 'does not register an offense' do
```

Comment:
It would be cool to add future, or current, support for well known other patterns like this

Reply:
Perhaps I should update the description on this one. Array.wrap is a Rails method, so I didn't want to draw too much attention to it. But it doesn't provide equivalent behavior because .wrap is a lot like `[*foo]` meaning when mapped it can take a list of Arrays and non-Arrays and put all the non-arrays into their own array while leaving the Arrays as is.
So I would suspect we would never want to change this, because we'd lose the safe autocorrect aspect of this Cop.

Reply:
Added [*foo] to the specs and made a context for this expected behavior